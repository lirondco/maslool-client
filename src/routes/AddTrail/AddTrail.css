.AddTrail {
    margin: 2%;
    padding: 2%;
}

.AddTrail h2 {
    margin-bottom: 1%;
}

hr {
    border: 1px solid;
}

.AddTrail_buttons {
    display: flex;
    flex-direction: column;
    margin-top: 3%;
}

.AddTrail_buttons button {
    margin-top: 2%;
}

