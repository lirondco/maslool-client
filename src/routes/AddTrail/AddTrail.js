import React, { Component } from 'react'
import EditTrailForm from '../../components/EditTrailForm/EditTrailForm'
import NewTrailForm from '../../components/NewTrailForm/NewTrailForm'
import { Button } from '../../components/Utils/Utils'
import './AddTrail.css'

export default class AddTrail extends Component {
    state = {
        status: null,
        message: 'Trail Options'
    }

    handleCancel = () => {
        this.setState({status: null, message: 'Trail Options'})
    }

    renderComponent = () => {
        if (this.state.status === 'editing') {
            return <EditTrailForm onCancelSuccess={this.handleCancel}/>
        } else if (this.state.status === 'adding') {
            return <NewTrailForm onCancelSuccess={this.handleCancel}/>
        }
    }

    handleAddMode = (ev) => {
        ev.preventDefault()
        this.setState({status: 'adding', message: 'Add Trail'})
        return
    }

    handleEditMode = (ev) => {
        ev.preventDefault()
        this.setState({status: 'editing', message: 'Edit Trail'})
        return
    }


    render() {
        return (
            <section className='AddTrail'>
                <h2>{this.state.message}</h2>
                <hr />
                {(this.state.status === null) ? <span className = 'AddTrail_buttons'><Button onClick={this.handleAddMode}>Add</Button><Button onClick={this.handleEditMode}>Edit</Button></span> : this.renderComponent()}
            </section>
        )
    }
}

