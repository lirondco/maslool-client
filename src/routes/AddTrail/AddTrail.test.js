import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddTrail from './AddTrail'

describe(`AddTrail route`, () => {
    it('renders the page', () => {
        const wrapper = shallow(<AddTrail />)
        expect(toJson(wrapper)).toMatchSnapshot()

    })

    it('First button works on click', () => {
        const wrapper = shallow(<AddTrail />)
        wrapper.find('ForwardRef').first().simulate('click', { preventDefault() {}})
    })

    it('Other button works on click', () => {
        const wrapper = shallow(<AddTrail />)
        wrapper.find('ForwardRef').last().simulate('click', { preventDefault() {}})
    })
})