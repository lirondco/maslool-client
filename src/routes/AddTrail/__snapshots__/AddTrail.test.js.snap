// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTrail route renders the page 1`] = `
<section
  className="AddTrail"
>
  <h2>
    Trail Options
  </h2>
  <hr />
  <span
    className="AddTrail_buttons"
  >
    <ForwardRef
      onClick={[Function]}
    >
      Add
    </ForwardRef>
    <ForwardRef
      onClick={[Function]}
    >
      Edit
    </ForwardRef>
  </span>
</section>
`;
